examples/fabric-bridge-app/linux/CommissionerControl.cpp

```diff
@@ -0,0 +1,204 @@
+    mVendorId     = request.vendorId;
+    mProductId    = request.productId;
+
+    if (request.label.HasValue())
+    {
+        const CharSpan & labelSpan = request.label.Value();
+        size_t labelLength         = labelSpan.size();
+
+        if (labelLength >= kLabelBufferSize)
+        {
+            ChipLogError(Zcl, "Label too long to fit in buffer");
+            return CHIP_ERROR_BUFFER_TOO_SMALL;
+        }
+
+        if (labelLength == 0)
+        {
+            mLabel.ClearValue();
+        }
+        else
+        {
+            memcpy(mLabelBuffer, labelSpan.data(), labelLength);
+            mLabelBuffer[labelLength] = '\0'; // Null-terminate the copied string
+            mLabel.SetValue(CharSpan(mLabelBuffer, labelLength));
+        }
+    }
+    else
+    {
+        mLabel.ClearValue();
+    }
+
+    return CommissionerControlServer::Instance().GenerateCommissioningRequestResultEvent(result);
+}
+
+CHIP_ERROR CommissionerControlDelegate::ValidateCommissionNodeCommand(NodeId clientNodeId, uint64_t requestId)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    // Verify if the CommissionNode command is sent from the same NodeId as the RequestCommissioningApproval.
+    VerifyOrExit(mClientNodeId == clientNodeId, err = CHIP_ERROR_WRONG_NODE_ID);
+
+    // Verify if the provided RequestId matches the value provided to the RequestCommissioningApproval.
+    VerifyOrExit(mRequestId == requestId, err = CHIP_ERROR_INCORRECT_STATE);
+
+exit:
+    return err;
+}
+
+CHIP_ERROR CommissionerControlDelegate::GetCommissioningWindowParams(CommissioningWindowParams & outParams)
+{
+    // Populate outParams with the required details.
+    outParams.iterations           = kIteration;
+    outParams.commissioningTimeout = kWindowTimeout;
+    outParams.discriminator        = kDiscriminator;
+
+    ReturnErrorOnFailure(Crypto::DRBG_get_bytes(mPBKDFSaltBuffer, sizeof(mPBKDFSaltBuffer)));
+    mPBKDFSalt     = ByteSpan(mPBKDFSaltBuffer);
+    outParams.salt = mPBKDFSalt;
+
+    Crypto::Spake2pVerifier verifier;
+    uint32_t setupPIN = kSetupPinCode;
+    ReturnErrorOnFailure(PASESession::GeneratePASEVerifier(verifier, kIteration, mPBKDFSalt, false, setupPIN));
+
+    MutableByteSpan serializedVerifierSpan(mPAKEPasscodeVerifierBuffer);
+    ReturnErrorOnFailure(verifier.Serialize(serializedVerifierSpan));
+    mPAKEPasscodeVerifier          = serializedVerifierSpan;
+    outParams.PAKEPasscodeVerifier = mPAKEPasscodeVerifier;
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR CommissionerControlDelegate::ReverseCommissionNode(const CommissioningWindowParams & params,
+                                                              const Optional<ByteSpan> & ipAddress, const Optional<uint16_t> & port)
+{
+#if defined(PW_RPC_FABRIC_BRIDGE_SERVICE) && PW_RPC_FABRIC_BRIDGE_SERVICE    
+    return ReverseCommissionBridge(Controller::CommissioningWindowVerifierParams()
+                                .SetNodeId(kUndefinedNodeId)
```

Comment:
Why are you setting the NodeId `kUndefinedNodeId`, can we add a comment as to why we are doing this here

Reply:
Removed, we don't use Node when commission a device, we don't need to set it.